### DISCUSSION TITLE
Is it possible to permanently set layer_ui_configs Locked property to True?

### DISCUSSION THREAD

**Discussion by ranga-tolapi** [pymupdf/PyMuPDF#2325] [Q&A]
If possible, please help to know how.

Below sample is taken from the documentation for layer_ui_configs.

```
>>> pprint(doc.layer_ui_configs())
 ({'depth': 0,
  'locked': True, # <===
  'number': 0,
  'on': True,
  'text': 'Circle',
  'type': 'checkbox'},
 {'depth': 0,
  'locked': False,
  'number': 1,
  'on': False,
  'text': 'Square',
  'type': 'checkbox'})
 >>>
```

**Comment by JorjMcKie** [comment]
This is a typical "Discussions" item, so I will transfer first.

**Reply by JorjMcKie** [reply]
You can always protect a PDF against changes by setting owner and user passwords (`owner_pw`/`user_pw`) and the permissions flags appropriately, see [here](https://pymupdf.readthedocs.io/en/latest/document.html#Document.save) and [here](https://pymupdf.readthedocs.io/en/latest/vars.html#document-permissions).

**Reply by ranga-tolapi** [reply]
My query is related to the Locked property of a layer. Please refer to the following screenshot.

![38963C93-69EF-4E61-A954-9D2DA396C8E1](https://user-images.githubusercontent.com/17163418/230729703-00a5d5c2-8f82-4ffa-98fe-08e871c696c0.jpeg)

**Reply by JorjMcKie** [reply]
My answer also was referring to this.
Have you read the PDF specification? "Locked" only means, that the user interface (Adobe or other viewer) cannot make changes.
Applications still can - so only password protect the PDF can prevent this. Remember you asked for "permanently" setting this.

**Reply by ranga-tolapi** [reply]
I want to set the Locked property to true for Adobe viewer to honour it (For now I’m okay if applications can make changes). How it can be done with PyMuPDF?

**Reply by JorjMcKie** [reply]
> I want to set the Locked property to true for Adobe viewer to honour it (For now I’m okay if applications can make changes). How it can be done with PyMuPDF?

This is not currently supported.

**Reply by JorjMcKie** [reply]
But we may consider extending the functionality to this.

**Reply by ranga-tolapi** [reply]
Noted, thank you @JorjMcKie!
